dvc/dependency/repo.py

```diff
@@ -72,7 +73,14 @@ def fetch(self):
 
             out = repo.find_out_by_relpath(self.def_path)
             with repo.state:
-                repo.cloud.pull(out.get_used_cache())
+                try:
+                    repo.cloud.pull(out.get_used_cache())
+                except NoRemoteError:
+                    # It would not be good idea to raise exception if the
+                    # file is already present in the cache
+                    if os.path.exists(self.repo.cache.local.get(out.checksum)):
```

Comment:
Let's use
```suggestion
                    if not self.repo.cache.local.changed_cache(out.checksum):
```
to make it fully work with directories and be more robust in general. There is nothing wrong with your implementation, just an obscure internal stuff that is more appropriate here.